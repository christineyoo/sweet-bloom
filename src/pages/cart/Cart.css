.cart-flex {
  display: flex;
  flex-direction: row;
}
.cart-flex-1 {
  flex: 1;
  border: 1px solid rgb(194, 39, 39);
  text-align: center;
}
.cart-flex-2 {
  flex: 2;
  border: 1px solid rgb(194, 39, 39);
  text-align: center;
}
