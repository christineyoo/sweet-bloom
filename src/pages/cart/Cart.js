import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import ItemCard from '../../organisms/itemCard/ItemCard';
import './Cart.css';

class Cart extends Component {
  state = {};
  render() {
    return (
      <div>
        <h1>Shopping cart</h1>
        <h2>Cart is empty</h2>
        <div className='cart-flex'>
          <section className='cart-flex-2'>
            <ItemCard />
            <ItemCard />
            <ItemCard />
            <ItemCard />
          </section>
          <section className='cart-flex-1'>
            <h1>Subtotal</h1>
            <h2>$200</h2>
            <h3>4 items</h3>
            <Link to='/checkout'>
              <button>Proceed to Checkout</button>
            </Link>
            <form>
              <h3>Do you want to add a message?</h3>
              <textarea
                rows='5'
                cols='30'
                placeholder='Gift message...'
              ></textarea>
              <p>
                <i>
                  Add your name to let the recipient know the gift is from you
                </i>
              </p>
              <button>Submit Message</button>
            </form>
          </section>
        </div>
      </div>
    );
  }
}

export default Cart;
