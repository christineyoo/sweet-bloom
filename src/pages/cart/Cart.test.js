import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Cart from './Cart';

it('renders Cart UI as expected', () => {
  const mockedParams = {
    itemsInCart: []
  };
  const tree = renderer
    .create(
      <Router>
        <Cart {...mockedParams} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
