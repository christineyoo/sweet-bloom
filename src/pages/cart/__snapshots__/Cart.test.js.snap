// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Cart UI as expected 1`] = `
<div
  id="cart"
>
  <h1>
    Shopping cart
  </h1>
  <p>
    <i>
      Do not refresh the page.
    </i>
  </p>
  <div
    className="cart-column-flex"
  >
    <h2>
      Cart is empty
    </h2>
  </div>
  <hr />
  <h3>
    Explore our shop
  </h3>
  <div
    className="cart-flex"
  >
    <section
      className="cart-flex-1"
    >
      <a
        href="/item/6"
        onClick={[Function]}
      >
        <img
          alt="loading..."
          height="300px"
          src="stock-photo-1.jpg"
        />
        <h3>
          Carnation Carnival
        </h3>
        <p>
          $25
        </p>
      </a>
    </section>
    <section
      className="cart-flex-1"
    >
      <a
        href="/item/20"
        onClick={[Function]}
      >
        <img
          alt="loading..."
          height="300px"
          src="stock-photo-2.jpg"
        />
        <h3>
          Tulip Town in Pink/Purple
        </h3>
        <p>
          $20
        </p>
      </a>
    </section>
    <section
      className="cart-flex-1"
    >
      <a
        href="/item/19"
        onClick={[Function]}
      >
        <img
          alt="loading..."
          height="300px"
          src="stock-photo-3.jpg"
        />
        <h3>
          Sunflowers
        </h3>
        <p>
          $15
        </p>
      </a>
    </section>
  </div>
</div>
`;
